packages/core/src/sql-on-fhir/eval.ts

```diff
@@ -59,12 +59,23 @@ function processResource(v: ViewDefinition, r: Resource): OutputRow[] {
     return [];
   }
 
+  const variables: Record<string, TypedValue> = {};
+  if (v.constant) {
+    for (const c of v.constant) {
+      const typedConstant = { type: 'ViewDefinitionConstant', value: c };
+      variables['%' + c.name] = getTypedPropertyValue(typedConstant, 'value') as TypedValue;
+    }
+  }
+
   const typedResource = toTypedValue(r);
 
   if (v.where) {
     for (const where of v.where) {
-      const whereResult = evalFhirPathTyped(where.path, [typedResource]);
-      if (whereResult.length !== 1 || whereResult[0].type !== 'boolean') {
+      const whereResult = evalFhirPathTyped(where.path, [typedResource], variables);
+      if (whereResult.length !== 1) {
+        return [];
```

Comment:
Should this throw an error instead of failing silently?

Reply:
Good question.  In this case, no, there are tests explicitly for each of these cases.

For example, there is a test that uses simply `Patient.active` and expects to gracefully handle the case where `Patient.active` is missing.